### PR TITLE
Introduce Operator.sol

### PR DESCRIPTION
**Pull request by se3000**
This PR also introduces support for Solidity v0.7, although it is marked as under development.

### CODE REVIEW

File: evm-contracts/src/dev_v0.7/Operator.sol [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+  /**
+   * @notice Displays the amount of LINK that is available for the node operator to withdraw
+   * @dev We use `ONE_FOR_CONSISTENT_GAS_COST` in place of 0 in storage
+   * @return The amount of withdrawable LINK on the contract
+   */
+  function withdrawable()
+    external
+    view
+    override(OracleInterface, WithdrawalInterface)
+    onlyOwner()
```
Comment (thodges-gh) [inline evm-contracts/src/dev_v0.7/Operator.sol:214]: While we're at it, can we get rid of `onlyOwner()` for this function?

**Comment by thodges-gh** [comment]
Another ask: can we throw everything in `v0.7/dev` instead of `dev_v0.7`?

**Comment by se3000** [comment]
> Another ask: can we throw everything in `v0.7/dev` instead of `dev_v0.7`?

I was running into some weirdness when things in `dev` were importing contracts in parent and peer directories. When we compile, it flattens everything into `abi/v*`, meaning no `abi/v0.6/dev`, everything goes into `abi/v0.6`. The imports in the contracts that are being recompiled get confused by this, and you have to blow away the `abi` directory before recompiling, which in turn slows down recompiling, and running tests, and the whole development cycle. We can put some effort into fixing up `belt` but it isn't a super quick fix. As a quick fix, I was planning on adding a `DEV_` prefix to contracts, but keeping the directory structure mostly flat, instead of having a `dev` subdirectory.

File: evm-contracts/src/dev_v0.7/Operator.sol [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+  function fulfillOracleRequest(
+    bytes32 requestId,
+    uint256 payment,
+    address callbackAddress,
+    bytes4 callbackFunctionId,
+    uint256 expiration,
+    bytes32 data
+  )
+    external
+    onlyAuthorizedNode
```
Comment (thodges-gh) [inline evm-contracts/src/dev_v0.7/Operator.sol:138]: We should make sure the order of modifiers fits Solidity's [style guide](https://solidity.readthedocs.io/en/latest/style-guide.html#function-declaration) recommendations:
1. Visibility
2. Mutability
3. Virtual
4. Override
5. Custom modifiers

File: evm-contracts/src/dev_v0.7/Operator.sol [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+import "./LinkTokenReceiver.sol";
+import "./Owned.sol";
+import "./interfaces/ChainlinkRequestInterface.sol";
+import "./interfaces/OracleInterface.sol";
+import "./interfaces/LinkTokenInterface.sol";
+import "./interfaces/WithdrawalInterface.sol";
+import "./vendor/SafeMath.sol";
+
+/**
+ * @title The Chainlink Oracle contract
```
Comment (thodges-gh) [inline evm-contracts/src/dev_v0.7/Operator.sol:12]:
```suggestion
 * @title The Chainlink Operator contract
```

File: evm-contracts/src/dev_v0.7/Operator.sol (hunk 2 above)
Comment (thodges-gh) [inline evm-contracts/src/dev_v0.7/Operator.sol:138]:
```suggestion
    onlyAuthorizedNode()
```
We should also always use `()` when no input params are needed for a modifier just to be consistent.

File: evm-contracts/src/dev_v0.7/Operator.sol [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+  /**
+   * @notice Allows the node operator to withdraw earned LINK to a given address
+   * @dev The owner of the contract can be another wallet and does not have to be a Chainlink node
+   * @param recipient The address to send the LINK token to
+   * @param amount The amount to send (specified in wei)
+   */
+  function withdraw(address recipient, uint256 amount)
+    external
+    override(OracleInterface, WithdrawalInterface)
+    onlyOwner
```
Comment (thodges-gh) [inline evm-contracts/src/dev_v0.7/Operator.sol:198]:
```suggestion
    onlyOwner()
```

**Pull request closed without merging**